src/aead.rs

```diff
@@ -140,29 +143,26 @@ pub struct Algorithm {
         aad: Aad<&[u8]>,
         in_out: &mut [u8],
         cpu_features: cpu::Features,
-    ) -> Tag,
+    ) -> Result<Tag, error::Unspecified>,
     open: fn(
         key: &KeyInner,
         nonce: Nonce,
         aad: Aad<&[u8]>,
         in_out: &mut [u8],
         src: RangeFrom<usize>,
         cpu_features: cpu::Features,
-    ) -> Tag,
+    ) -> Result<Tag, error::Unspecified>,
 
     key_len: usize,
     id: AlgorithmID,
-
-    /// Use `max_input_len!()` to initialize this.
-    // TODO: Make this `usize`.
-    max_input_len: u64,
 }
 
-const fn max_input_len(block_len: usize, overhead_blocks_per_nonce: usize) -> u64 {
+const fn max_input_len(block_len: usize, overhead_blocks_per_nonce: usize) -> usize {
     // Each of our AEADs use a 32-bit block counter so the maximum is the
```

Comment:
Codecov complains that this is not covered by tests, but it is only used in const contexts, so that is OK.